api/envoy/config/cluster/v3/cluster.proto

```diff
@@ -1005,6 +1005,12 @@ message Cluster {
   // If `connection_pool_per_downstream_connection` is true, the cluster will use a separate
   // connection pool for every downstream connection
   bool connection_pool_per_downstream_connection = 51;
+
+  // If `erase_idle_pools` is true, the cluster will remove connection pools
+  // that no longer have active connections. The time between the last activity
+  // in a pool and its cleanup can be adjusted via the cluster's connection
+  // :ref:`idle_timeout <envoy_api_field_config.cluster.v3.Cluster.common_http_protocol_options>`
+  bool erase_idle_pools = 53;
```

Comment:
What is the actual cost of adding a new pool? I'm thinking not that much? And if we only remove a pool in the case of connection idle timeouts or failure of all connections to all hosts, it seems like an edge case and not a huge deal? I think my preference would be to remove this config option, runtime guard the change per your recommendation, but default it to on? @alyssawilk wdyt?